### PR TITLE
FEATURE: Allow complex post params from plugin

### PR DESCRIPTION
**Pull request by markvanlan**
This allows hashes and arrays to be permitted into the create post params

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -684,8 +684,16 @@ def create_params
       :draft_key
     ]
 
-    Post.plugin_permitted_create_params.each do |key, plugin|
-      permitted << key if plugin.enabled?
+    Post.plugin_permitted_create_params.each do |key, value|
```
Comment (ZogStriP) [inline app/controllers/posts_controller.rb]: Could you add some tests to ensure this API does not regress?

Comment (markvanlan) [reply app/controllers/posts_controller.rb]: @eviltrout Tests have been added

File: lib/plugin/instance.rb [hunk 2]
```diff
@@ -250,9 +250,9 @@ def add_preloaded_topic_list_custom_field(field)
   end
 
   # Add a permitted_create_param to Post, respecting if the plugin is enabled
-  def add_permitted_post_create_param(name)
+  def add_permitted_post_create_param(name, type = "")
```
Comment (eviltrout) [inline lib/plugin/instance.rb]: This API is a bit odd to me. You pass an object of the type you want?

I would prefer an enum of something or a symbol.

Comment (markvanlan) [reply lib/plugin/instance.rb]: @eviltrout I made the change to use symbols. The internals look a lot better, but the api is less consistent now with how strong params are handled in controllers.

**Pull request merged**